tests/pagination.tsx

```diff
@@ -0,0 +1,135 @@
+import * as enzyme from 'enzyme';
+import * as React from 'react';
+import * as renderer from 'react-test-renderer';
+
+import { Pagination } from '../src/ts';
+
+describe('Pagination', () => {
+  it('should match snapshot with current page button highlighted', () => {
+    const instance = renderer.create(
+      <Pagination
+        className="float-right margin-top-base"
+        pageSize={3}
+        changePage={jest.fn()}
+        currentPageNumber={1}
+        itemCount={26}
+      />
+    );
+
+    expect(instance).toMatchSnapshot();
+  });
+
+  it('should disable buttons when disabled is true', () => {
+    const instance = renderer.create(
+      <Pagination
+        pageSize={3}
+        changePage={jest.fn()}
+        currentPageNumber={1}
+        itemCount={26}
+        disabled
+      />
+    );
+
+    expect(instance).toMatchSnapshot();
+  });
+
+  it('should render the first page link as an first option', () => {
+    const instance = renderer.create(
+      <Pagination
+        pageSize={3}
+        changePage={jest.fn()}
+        currentPageNumber={5}
+        itemCount={70}
+      />
+    );
+
+    expect(instance).toMatchSnapshot();
+  });
+
+  it('should display less than 5 buttons for minimal page counts', () => {
+    const instance = renderer.create(
+      <Pagination
+        pageSize={3}
+        changePage={jest.fn()}
+        currentPageNumber={1}
+        itemCount={6}
+      />
+    );
+
+    expect(instance).toMatchSnapshot();
+  });
+
+  it('should display custom next and previous text', () => {
+    const instance = renderer.create(
+      <Pagination
+        pageSize={3}
+        changePage={jest.fn()}
+        currentPageNumber={1}
+        itemCount={6}
+        prevText="Prev"
+        nextText="Next"
+      />
+    );
+
+    expect(instance).toMatchSnapshot();
+  });
+
+  it('should add an extra button to cover remainder pages', () => {
```

Comment:
I'm not actually sure what this is testing.

Reply:
It may not be necessary to test this, but it renders the 'next button' for remainders. For example, the page size is 3 but the total items are 7, so it renders buttons and extra for the remainder for 1. `7%3=1` soz

Reply:
```suggestion
  it('should add a "next" button when there are more than 3 pages', () => {
```